pkg/utils/cluster.go

```diff
@@ -16,8 +17,22 @@ type BackplaneCluster struct {
 	BackplaneHost string // for e.g. https://api-backplane.apps.com
 }
 
+type ClusterUtils interface {
+	GetClusterIDAndHostFromClusterURL(clusterURL string) (string, string, error)
+	GetBackplaneClusterFromConfig() (BackplaneCluster, error)
+	GetBackplaneClusterFromClusterKey(clusterKey string) (BackplaneCluster, error)
+	GetCloudProvider(cluster *cmv1.Cluster) string
+	GetBackplaneCluster(params ...string) (BackplaneCluster, error)
+}
+
+type DefaultClusterUtilsImpl struct {}
+
+var (
+	DefaultClusterUtils ClusterUtils = &DefaultClusterUtilsImpl{}
+)
+
```

Comment:
This is really good and it allows mocking more unit tests :+1:  :1st_place_medal:

Reply:
Thanks @samanthajayasinghe ! Ready for another review now I think